test/metabase/upload_test.clj

```diff
@@ -169,17 +169,23 @@
            [" 2022-01-01 "                    #t "2022-01-01"             date-type]
            [" 2022-02-30 "                    " 2022-02-30 "              vchar-type]
            [" -2022-01-01 "                   #t "-2022-01-01"            date-type]
+           [" Jan 1 2018"                     #t "2018-01-01"             date-type]
            [" Jan 30 2018"                    #t "2018-01-30"             date-type]
            [" Jan 30 -2018"                   #t "-2018-01-30"            date-type]
            [" Jan 30, 2018"                   #t "2018-01-30"             date-type]
            [" Feb 30, 2018"                   " Feb 30, 2018"             vchar-type]
+           [" 1 Jan 2018"                     #t "2018-01-01"             date-type]
            [" 30 Jan 2018"                    #t "2018-01-30"             date-type]
            [" 30 Jan, 2018"                   #t "2018-01-30"             date-type]
+           [" January 1 2018"                 #t "2018-01-01"             date-type]
```

Comment:
we need a test for case where the day has a leading zero, like `January 01 2018`

Reply:
Good point, covered here [f4bc5df](https://github.com/metabase/metabase/pull/38276/commits/f4bc5df1e621c469cda12553e51d9c234998a5ec)